File: 2123.py
def counter(arr):
    count = 0
    avg = sum(arr) // len(arr)
    if sum(arr) % len(arr):
        print("-1")
    else:
        for elem in arr:
            if elem < avg:
                count += avg - elem
        print(count)


def main():
    arr = []
    while(True):
        n = int(input())
        if(n == -1):
            break
        else:
            while(n):
                arr.append(int(input()))
                n -= 1
            counter(arr)


if __name__ == '__main__':
    main()

File: 2123.cpp
/*
   CANDY - Candy I
 */

#include <iostream>

using namespace std;


int main(){
        int n;
        cin >> n;
        while(n != -1 ) {
                int arr[n];
                int sum=0,count=0,avg;

                for(int i=0; i<n; i++) {
                        cin >> arr[i];
                        sum += arr[i];
                }
                avg = sum/n;
                if (sum%n == 0) {
                  for(int i=0; i<n; i++) {
                          if(arr[i]<avg) {
                                  count += avg-arr[i];
                          }
                  }
                  cout << count << endl;
                  cin >> n;
                } else {
                  cout << "-1" << endl;
                  cin >> n;
                }

        }
        return 0;
}
